app/src/main/java/com/kickstarter/libs/utils/extensions/AnyExt.kt

```diff
@@ -0,0 +1,54 @@
+package com.kickstarter.libs.utils.extensions
+
+import com.kickstarter.libs.utils.ObjectUtils
+import io.reactivex.functions.Function
+import rx.functions.Func1
+
+fun <T : Any> T?.isNull(): Boolean {
+    return this == null
+}
+
+fun <T : Any> T?.isNotNull(): Boolean {
+    return this != null
+}
+
+fun <T : Any> T?.coalesce(theDefault: T): T {
+    if (this.isNotNull()) {
+        return this!!
+    }
+    return theDefault
+}
+
+fun <T : Any> coalesceWith(theDefault: T): Func1<T, T> {
+    return Func1 { value: T -> ObjectUtils.coalesce(value, theDefault) }
+}
+
+fun <T : Any> coalesceWithV2(theDefault: T): Function<T, T> {
+    return Function { value: T -> ObjectUtils.coalesce(value, theDefault) }
+}
+
+fun <T : Any> T?.numToString(): String? {
+    return when (this) {
+        is Long -> this.toString()
+        is Float -> this.toString()
+        is Int -> this.toString()
+        is Double -> this.toString()
+        else -> null
+    }
+}
+
+fun <T : Any> T?.requireNonNull(): T {
```

Comment:
I think all of these requireNonNull methods can be removed in favor of `Preconditions.kt requireNotNull`

Reply:
correct, kotlin already has a [`requireNotNull()`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/require-not-null.html#requirenotnull), as example take a look at `ProjectPageViewModel` every `requireNotNull` would be the `Preconditions` one, so for any kotlin code use kotlin available precondition and not  `ObjectUtils.requireNotNull()`. 

So I suggest a quick search over the project and exchange any  `ObjectUtils.requireNotNull()` over the precondition instead of the extension methods. And remove the extension methods and possible `ObjectUtils.requireNotNull()`  as well (We can keep the latest one in java code if needed).

In `ProjectCardViewHolder.kt` we could:
`
        val projectAndParams = requireNotNull(data as? Pair<Project, DiscoveryParams>)
`
instead of using the extension method: 
`
        val projectAndParams = (data as? Pair<Project, DiscoveryParams>).requireNonNull()
`

Reply:
I ended up keeping the AnyExt require non-null since there is a java class currently using it, ActivityListViewHolder. Once we migrate that to java or deprecate it, we can remove these extension methods in favor of the kotlin `Preconditions.requireNotNull()`